internal/gatewayapi/testdata/httproute-with-conflicting-filters.in.yaml

```diff
@@ -0,0 +1,90 @@
+gateways:
+  - apiVersion: gateway.networking.k8s.io/v1
+    kind: Gateway
+    metadata:
+      namespace: envoy-gateway
+      name: gateway-1
+    spec:
+      gatewayClassName: envoy-gateway-class
+      listeners:
+        - name: http
+          protocol: HTTP
+          port: 80
+          hostname: "*.envoyproxy.io"
+          allowedRoutes:
+            namespaces:
+              from: All
+httpRoutes:
+  - apiVersion: gateway.networking.k8s.io/v1
+    kind: HTTPRoute
+    metadata:
+      namespace: default
+      name: mirror-direct-response
+    spec:
+      hostnames:
+        - gateway.envoyproxy.io
+      parentRefs:
+        - namespace: envoy-gateway
+          name: gateway-1
+          sectionName: http
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: /
+          backendRefs:
+            - name: service-1
+              port: 8080
+          filters:
+            - type: RequestMirror
+              requestMirror:
+                backendRef:
+                  kind: Service
+                  name: service-1
+                  port: 8080
+            - type: ExtensionRef
+              extensionRef:
+                group: gateway.envoyproxy.io
+                kind: HTTPRouteFilter
+                name: mirror-direct-response
+  - apiVersion: gateway.networking.k8s.io/v1
+    kind: HTTPRoute
+    metadata:
+      namespace: default
+      name: mirror-request-redirect
+    spec:
+      hostnames:
+        - gateway.envoyproxy.io
+      parentRefs:
+        - namespace: envoy-gateway
+          name: gateway-1
+          sectionName: http
+      rules:
+        - matches:
```

Comment:
can we add a mixed rule case, where one rule is valid and one isnt, to make sure the valid rule is not impacted

Reply:
According to the Gateway API specific, The HTTPRoute`Accepted` condition should be set to `false` with `IncompatibleFilters` reason when there are incompatible filters present on a route rule.

https://github.com/kubernetes-sigs/gateway-api/blob/f24f3a61f398c65ab629da1843cb65fd5ec9419f/apis/v1/shared_types.go#L384-L387

And EG skips the HTTPRoutes with `Accepted` status as `False`, which is reasonable as the entire HTTPRoute has been rejected. 

https://github.com/envoyproxy/gateway/blob/36a23a6377ee7711164a872241201f96324a0498/internal/gatewayapi/route.go#L147-L149

Changing this behavior would require a global update and could affect other cases as well.